src/diff_edit.rs

```diff
@@ -52,6 +54,31 @@ pub enum DiffEditError {
     IoError(#[source] std::io::Error),
     #[error("Failed to snapshot changes: {0:?}")]
     SnapshotError(SnapshotError),
+    #[error("Couldn't find the path {0:?} in this revision")]
```

Comment:
I think I'd prefer to see these errors in a separate type since they can't happen in `edit_diff()`. It feels like all the mergetool-related code should be in a separate module, or how much of this code is reused between the `run_mergetool()` and `edit_diff()`?

Reply:
They get moved to a separate type in another commit (I meant to squash it in here, but didn't get to it).

The configuration code is heavily shared between the two functions, but not much else.